### ISSUE TITLE
Consider using pdqsort to get higher performance?

### ISSUE THREAD

**Issue by Aloxaf** [kimci86/bkcrack#2]
I noticed that [`std::sort`](https://github.com/kimci86/bkcrack/blob/master/src/Zreduction.cpp#L51) takes up most of the time on Zreduction.

By simply replacing `std::sort` with [`pdqsort`](https://github.com/orlp/pdqsort), we can get  higher performance.

**Comment by kimci86** [comment]
Hi, thank you for suggesting this improvement.

I also tried pdqsort some time ago, but it looked to me that the improvement was small on bkcrack. Did you benchmark it?

If time really matters, GCC also provides a parallel mode with a parallelized implementation of sort. See [parallel mode documentation](https://gcc.gnu.org/onlinedocs/libstdc++/manual/parallel_mode.html).

I could provide pdqsort or GCC parallel sort as an option, but I will first investigate, soon™.
Also, feel free to open a pull request if you find the improvement is worth it.

**Comment by kimci86** [comment]
As the Z-reduction step usually does not last more than a couple of seconds, I did not put much effort in it, but your suggestion convinced me to work on that part.
A nice side effect is that I found a simple improvement of the algorithm that cuts off about 25% of the time spent in reduction. In addition pdqsort gives a nice boost for a total decrease of about 40%.
That was worth it! See https://github.com/kimci86/bkcrack/compare/1b82756cdf1452ae4efdb6acf80d3dac888a83c3...827f72a0b9ef693c4f29b1221aea1c896ad9446c
